### PR TITLE
Extended list of sensitive configuration values

### PR DESCRIPTION
**Pull request by Fiech**
In accordance with the issuetemplate app issue:
https://github.com/nextcloud/issuetemplate/issues/27

Signed-off-by: Johannes Schlichenmaier <johannes@schlichenmaier.info>

### CODE REVIEW

**Review by juliusknorr — APPROVED** [review]

**Review by rullzer — APPROVED** [review]
Looks good to me

**Comment by rullzer** [comment]
Thanks a lot @Fiech

**Comment by Fiech** [comment]
No worries.

One question, though: How is this code coverage report to be interpreted?

File: lib/private/SystemConfig.php [hunk 1]
```diff
@@ -37,22 +37,30 @@ class SystemConfig {
 
 	/** @var array */
 	protected $sensitiveValues = [
+		'instanceid' => true,
+		'trusted_domains' => true,
+		'datadirectory' => true,
+		'overwrite.cli.url' => true,
```
Comment (nickvergessen) [inline lib/private/SystemConfig.php]: Please revert this and trusted domains, its actually the number one fault on most instances which have problems with links

Comment (Fiech) [reply lib/private/SystemConfig.php]: Sorry, which one? All 4 lines?

In https://github.com/nextcloud/issuetemplate/issues/27 we talked about the problem of sensitive data being automatically put into issue reports.

I personally would not necessarily want - for example - the address my cloud server is running on to be put in an issue, if it's not relevant to my problem.

Should it not be up to the user, to provide this information? Although, we can maybe put a warning into the issuetemplate, or a checkbox or something? Then we'd have to reopen the issue again...

Comment (juliushaertl) [reply lib/private/SystemConfig.php]: Maybe we should then make the filter for those values available as an option in the issuetemplate app. Another option would be to just filter out the domain part of the values, so we can see there is some option configured but replace the domain with some generic value.

Comment (Fiech) [reply lib/private/SystemConfig.php]: Good idea,

so, it's about the fields `overwrite.cli.url` and the `trusted_domains` array?

How would I go about reverting this within the process? Should I amend and force push my last commit? Will this retrigger the pull request, or should I make a new pull request with a new commit?

Can you reopen the issue then?

Comment (nickvergessen) [reply lib/private/SystemConfig.php]: Yeah e.g. most common problem is overwrite.cli.url without protocol or path, etc. When the value is filtered out it just means we have to ask for the info anyway.

Comment (nickvergessen) [reply lib/private/SystemConfig.php]: And trusted domain with either of those...

**Pull request merged**